### ISSUE TITLE
AutoComplete: Suggestion box doesn't mark searched text

### ISSUE THREAD

**Issue by milansie** [primefaces/primefaces#11012]
### Describe the bug

Hello,

since this change https://github.com/primefaces/primefaces/issues/10397 (13.0.1? version) marking searched text (bold) in autocomplete suggestion box doesn't work anymore.

I understand the need of that change, but this solution doesn't work.  

Check the source code, it doesn't make any sense:
```     
                    var re = new RegExp('(' + queryParts.join('|') + ')', 'gi');
.....
                    queryResults.filter(':not(.ui-autocomplete-moretext)').each(function() {
                        var item = $(this);
                        var escape = $this.cfg.escape;
                        var text = escape ? item.html() : item.text();
                        var escaped_re = escape ? /${PrimeFaces.escapeHTML(text)}/g : /${text}/g;
                        item.html(text.replace(escaped_re, '<span class="ui-autocomplete-query">$&</span>'));
```

Typed text (re variable) doesn't affect replacing code in any way.

In previous version was regexp (from typed text) replaced by span with class -  and that worked. In new version there is replacing text (suggestion item text) with .. ehm .. some weird regexp. What is the purpose of that? I think that piece of code doesn't work in any case.

I think, that it should be something like this:
```
var escaped_text =  escape ? PrimeFaces.escapeHTML(text) : text;
item.html(escaped_text.replace(re,  '<span class="ui-autocomplete-query">$&</span>'));
```

But that doesn't solve problem from  https://github.com/primefaces/primefaces/issues/10397 . I don't know, if there is any possible solution for that issue (yes, this change "fix" the issue, that replacing doesn't break the code - but marking the search text doesn't work in any way).

I think, there is no solution for marking search text in suggestion box, while using advanced suggestion box with p:column and html tags. You can't avoid replacing parts of html tags by span with bold class without breaking the html. So I personally think, that best solution for this is to remove this feature for advanced suggestion box with html tags. But maybe you could think of another solution.

thank you very much
milansi

### Reproducer

https://github.com/milansie/pf_autocomplete_sg_bug

### Expected behavior

Type text in autocomplete is marked bold in solution (same way as it works on primefaces showcase)

### PrimeFaces edition

None

### PrimeFaces version

13.0.1+

### JSF implementation

All

### JSF version

4.0

### Java version

17+

**Comment by melloware** [comment]
Your reproducer doesn't work with `mvn clean jetty:run` ? Did you use PrimeFaces Test?

**Comment by melloware** [comment]
Reproducer: 
[pf-11012-jetty.zip](https://github.com/primefaces/primefaces/files/13442313/pf-11012-jetty.zip)

**Comment by milansie** [comment]
> Reproducer: [pf-11012-jetty.zip](https://github.com/primefaces/primefaces/files/13442313/pf-11012-jetty.zip)

I added jetty support to pom.xml, but I see you solved it already. thanks

**Comment by melloware** [comment]
MonkeyPatch:

```js
/* Place any Javascript here you want loaded in test.xhtml */
if (PrimeFaces.widget.AutoComplete) {

    PrimeFaces.widget.AutoComplete.prototype.showSuggestions = function(query) {
        this.items = this.panel.find('.ui-autocomplete-item');
        this.items.attr('role', 'option');

        if (this.cfg.grouping) {
            this.groupItems();
        }

        this.bindDynamicEvents();

        var $this = this,
            hidden = this.panel.is(':hidden');

        if (hidden) {
            this.show();
        } else {
            this.alignPanel();
        }

        // #8717 always clear list before trying to fill it
        if (this.cfg.forceSelection) {
            this.currentItems = [];
        }

        if (this.items.length > 0) {
            var firstItem = this.items.eq(0);

            //highlight first item
            if (this.cfg.autoHighlight && firstItem.length) {
                this.highlightItem(firstItem, true);
                this.changeAriaValue(firstItem[0]);
            }

            //highlight query string
            if (query.length > 0) {
                var cleanedQuery = query.trim().replaceAll(/(\s+)/g, ' ');
                if (cleanedQuery.length > 0) {
                    var queryParts = cleanedQuery.split(' ');
                    for (var i = 0; i < queryParts.length; i++) {
                        queryParts[i] = PrimeFaces.escapeRegExp(queryParts[i]);
                    }
                    var re = new RegExp('(' + queryParts.join('|') + ')', 'gi');
                    var isCustomContent = this.panel.children().is('table');
                    var queryResults = isCustomContent ? this.panel.children().find('span') : this.items;
                    queryResults.filter(':not(.ui-autocomplete-moretext)').each(function() {
                        var item = $(this);
                        var escape = $this.cfg.escape;
                        var text = escape ? item.html() : item.text();
                        var escaped_re = escape ? /${PrimeFaces.escapeHTML(text)}/g : /${text}/g;
                        text = text.replace(escaped_re, '$&');
                        item.html(text.replace(re, '<span class="ui-autocomplete-query">$&</span>'));
                    });
                }
            }

            if (this.cfg.forceSelection) {
                this.items.each(function(i, item) {
                    $this.currentItems.push($(item).attr('data-item-label'));
                });
            }

            //show itemtip if defined
            if (this.cfg.autoHighlight && this.cfg.itemtip && firstItem.length === 1) {
                this.showItemtip(firstItem);
            }

            this.displayAriaStatus(this.resultsMessage.replace('{0}', this.items.length));
        } else {
            if (this.emptyMessage && this.cfg.forceSelection) {
                var emptyText = '<div class="ui-autocomplete-empty-message ui-widget">' + PrimeFaces.escapeHTML(this.emptyMessage) + '</div>';
                this.panel.prepend(emptyText);
            } else if (!this.cfg.hasFooter) {
                this.panel.hide();
            }

            this.input.removeAttr('aria-activedescendant');
            this.displayAriaStatus(this.emptyMessage);
        }
    }
};
```

**Comment by melloware** [comment]
Fixed for 13.0.4
